### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by GoBenGoh**
Txt file bug fix + comments and clean up

### CODE REVIEW

**Comment by GoBenGoh** [comment]
Do not approve yet! Need to fix TextToJava!

File: LLM Testing/Code-Testing/src/main/java/App.java [hunk 1]
```diff
@@ -5,32 +5,33 @@
 import java.util.List;
 
 public class App {
-    public static void main(String[] args) throws IOException, GitAPIException {
+    public static void main(String[] args) throws IOException, GitAPIException, InterruptedException {
```
Comment (cmrnnathan) [inline LLM Testing/Code-Testing/src/main/java/App.java:8]: Just wondering what the InterruptedException is for

File: LLM Testing/Code-Testing/src/main/java/App.java [hunk 2]
```diff
@@ -5,32 +5,33 @@
...
+//            List<String> repoCommits = repoHandler.getAllCommitHashes();
+//            for (String commitHash: repoCommits) {
+//                repoHandler.switchToCommit(commitHash);
+//                ChatGPTAPI.runTestIterations(args, studentRepo, commitHash, workflow);
+//            }
+//
+//            // Switch back to the initial commit before closing the repository
+//            repoHandler.switchToCommit(repoCommits.get(0));
+//            repoHandler.close();
+//        }
```
Comment (cmrnnathan) [inline LLM Testing/Code-Testing/src/main/java/App.java]: Why is this section commented out?

File: LLM Testing/Code-Testing/src/main/java/FailureFileWriter.java [hunk 3]
```diff
@@ -40,4 +40,19 @@ public static void writeFailuresToFile(List<String> failedProvidedTests, List<St
...
+        failedTests.addAll(failedHiddenTests);
+        List<String> assertionMessages = new ArrayList<>(providedAsserts);
+        assertionMessages.addAll(hiddenAsserts);
+        StringBuffer failures = new StringBuffer();
+        for (int i = 0; i < failedTests.size(); i++) {
+            String failedTest = failedTests.get(i);
+            failures.append(failedTest);
+            failures.append("\n");
+            String assertionMessage = assertionMessages.get(i);
+            failures.append(assertionMessage);
```
Comment (cmrnnathan) [inline LLM Testing/Code-Testing/src/main/java/FailureFileWriter.java:54]: What does the output of this look like? Would it also need another new line statement at the end after ```failures.append(assertionMessage);```

**Review by cmrnnathan — COMMENTED** [review]
Nice. I just had a couple of questions

File: LLM Testing/Code-Testing/src/main/java/App.java (hunk 2 above)
Comment (GoBenGoh) [reply LLM Testing/Code-Testing/src/main/java/App.java]: For testing purposes. I put it back.

File: LLM Testing/Code-Testing/src/main/java/App.java (hunk 1 above)
Comment (GoBenGoh) [reply LLM Testing/Code-Testing/src/main/java/App.java:8]: Forgot to remove it. Fixed.

File: LLM Testing/Code-Testing/src/main/java/FailureFileWriter.java (hunk 3 above)
Comment (GoBenGoh) [reply LLM Testing/Code-Testing/src/main/java/FailureFileWriter.java:54]: Added the new line

**Review by cmrnnathan — APPROVED** [review]
Looks good.

**Pull request merged**